### DISCUSSION TITLE
Akka seed project

### DISCUSSION THREAD

**Discussion by ghost** [Sunbird-Lern/Community#93] [Installation]
Hello, 

I am trying to run notification service from Lern module. Following the steps from read me file, I was able to build and start the service but akka seed project without router was not there.

Can you help me in explaining the steps for creating Akka seed project? As per the screenshot below
 
<img width="809" alt="notificationReadMe" src="https://user-images.githubusercontent.com/126251863/227153160-f56cf626-503f-4f16-95ec-4a72a94ce523.png">

**Comment by reshmi-nair** [comment]
@ravi-kant-saini I will check on this and update soon

**Reply by reshmi-nair** [reply]
@ravi-kant-saini You can check some sample projects here [play-service-seeder](https://github.com/indrajra/play-service-seeder)
There is with router and without router implementation. 
Notification-service is based on play-akka without router implementation. You can refer to application.conf and routes files in service folder to check the actor and request-controller mappings respectively

**Comment by ghost** [comment]
Hi, 
I was trying to build play service seeder without router in which there are some pre-requisites as shown in screenshot attached.  As I am new to this play framework, I request you to guide me about the environment vars needed as sample setVars.sh file is empty. And, another step is to update the key value in application.conf but there is no field for key in the said file
. 
I would also like to know that once I have build my play seeder, how do I integrate our sunbird notification service with it?

Thanks

<img width="960" alt="setVars" src="https://user-images.githubusercontent.com/126251863/231733655-bd65665f-953a-4589-a936-cdab9f002d20.png">
<img width="960" alt="pre-requisites" src="https://user-images.githubusercontent.com/126251863/231733664-4ff2f2a1-0bf7-43f5-ad57-6543f0d31806.png">

**Reply by reshmi-nair** [reply]
@ravi-kant-saini When you are trying to build a new service with play-seeder implementation, there will not be any env variables, until your logic needs some variables.

In application.conf, "thisActorSystem" variable will be present only in case of "service-seed-with-router" option.

**Reply by ghost** [reply]
@reshmi-nair Thank you for your response, I was able to build play project. Coming to Sunbird notification service, I am facing error with sb-actor, what could be the issue here?
<img width="956" alt="actor" src="https://user-images.githubusercontent.com/126251863/232414176-6d8fb4f8-29e9-464c-913b-b884aa95a064.png">

**Reply by reshmi-nair** [reply]
@ravi-kant-saini It seems like a problem with java version. Pls check this link for reference [ref](https://mkyong.com/maven/maven-error-invalid-target-release-1-11/).

**Reply by ghost** [reply]
@reshmi-nair Above issue was fixed. Now, when I am trying to execute last step from readme file for Notification service which is mvn play2:start; I am getting below error :
Failed to execute goal com.google.code.play2-maven-plugin:play2-maven-plugin:1.0.0-rc5:start (default-cli) on project notification-service: Play! Server already started ("RUNNING_PID" file found)
<img width="956" alt="startFail" src="https://user-images.githubusercontent.com/126251863/233017427-dac0bfe8-5534-4108-8429-93dd9bd35827.png">

**Reply by reshmi-nair** [reply]
@ravi-kant-saini Please use the command mvn play2:run to run the service

**Reply by ghost** [reply]
Hi @reshmi-nair , I started the app using mvn play2:run and it is giving an exception as below:
Unexpected exception
CreationException: Unable to create injector, see the following errors:

1) Error injecting constructor, org.sunbird.common.exception.BaseException: All host(s) tried for query failed (tried: localhost/127.0.0.1:9042 (com.datastax.driver.core.exceptions.TransportException: [localhost/127.0.0.1:9042] Cannot connect), localhost/0:0:0:0:0:0:0:1:9042 (com.datastax.driver.core.exceptions.TransportException: [localhost/0:0:0:0:0:0:0:1:9042] Cannot connect))
  at utils.ApplicationStart.<init>(ApplicationStart.java:32)
  at utils.module.StartModule.configure(StartModule.java:16) (via modules: com.google.inject.util.Modules$OverrideModule -> utils.module.StartModule)
  while locating utils.ApplicationStart
<img width="956" alt="exception" src="https://user-images.githubusercontent.com/126251863/233393311-a7ab6e19-a34f-49c7-82bb-55677a05d055.png">

**Reply by Jayaprakash8887** [reply]
@ravi-kant-saini Seems like cassandra connection is failing. Kindly verify cassandra.

**Comment by ghost** [comment]
Solved, Thanks team.
